i18n_scripts/combine_translations_files.py

```diff
@@ -0,0 +1,105 @@
+from collections import OrderedDict
+
+
+def translation_file_of(module):
+    translation_file = os.path.join(modules_dir, module, module, 'en.lproj', 'Localizable.strings')
+    return translation_file
+
+def get_modules_to_translate(modules_dir):
+    """
+    Retrieve the names of modules that have translation files for a specified language.
+
+    Parameters:
+        modules_dir (str): The path to the directory containing all the modules.
+
+    Returns:
+        list of str: A list of module names that have translation files for the specified language.
+    """
+    dirs = [directory for directory in os.listdir(modules_dir) if
+            os.path.isdir(os.path.join(modules_dir, directory))]
+
+    modules_list = []
+    for module in dirs:
+        translation_file = translation_file_of(module)
+        if os.path.exists(translation_file) and os.path.isfile(translation_file):
+            modules_list.append(module)
+    return modules_list
+
+
+def get_translations(modules_dir):
+    """
+    Retrieve the list of translations from all specified modules.
+
+    Parameters:
+        modules_dir (str): The directory containing the modules.
+
+    Returns:
+        OrderedDict of dict: An ordered dict of dictionaries containing the 'key', 'value', and 'comment' for each
+        translation line. The key of the OrderedDict is the value of the key in the contained dict in addition to the
+        name of the module containing the translation
+    """
+    ordered_dict_of_translations = OrderedDict()
+    modules = get_modules_to_translate(modules_dir)
+    for module in modules:
+        translation_file = translation_file_of(module)
+        module_translations = localizable.parse_strings(filename=translation_file)
+
+        for line in module_translations:
+            ordered_dict_of_translations[f"{module}.{line['key']}"] = line
+
+    return ordered_dict_of_translations
+
+
+def _escape(s):
+    """
+    Reverse the replacements performed by _unescape() in the localizable library
+    """
+    s = s.replace('\n', r'\n').replace('\r', r'\r').replace('"', r'\"')
+    return s
+
+
+def write_dict_to_file(content_ordered_dict, modules_dir):
+    """
+       Write the contents of an ordered dictionary to a Localizable.strings file.
+
+       This function takes an ordered dictionary containing translation data and writes it to a Localizable.strings
+       file located in the 'I18N/en.lproj' directory within the specified modules directory. It creates the directory
+       if it doesn't exist.
+
+       Parameters:
+           content_ordered_dict (OrderedDict): An ordered dictionary containing translation data. The keys
+           are the translation keys, and the values are dictionaries with 'value' and 'comment' keys representing the
+           translation value and optional comments, respectively.
+           modules_dir (str): The path to the modules directory
+           where the I18N directory will be written.
```

Comment:
Please de-indent by 3-spaces to match the start of `"""`

Reply:
done